### PR TITLE
Migrate test in `pickles_base` to alcotest

### LINKED ISSUE #13710: Move `pickles_base` tests to `alcotest`

### PR DESCRIPTION
**Pull request by rbonichon**
Remove the use of inline tests in `pickles_base` and use alcotest instead.

This makes the output nicer and allows to monitor progress of the testsuite.
But first, and foremost, it enforces a clean separation between test code and implementation code.

Fix #13710

### CODE REVIEW

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me

File: src/lib/pickles_base/test/dune [hunk 1]
```diff
@@ -0,0 +1,15 @@
+(tests
```
Comment (dannywillems) [inline src/lib/pickles_base/test/dune:1]: can we add bisect_ppx? even if we do not use it at the moment.

**Review by dannywillems — APPROVED** [review]

**Comment by rbonichon** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Pull request merged**